zbus/src/match_rule/mod.rs

```diff
@@ -8,9 +8,12 @@ use zvariant::Structure;
 use crate::{
     names::{BusName, InterfaceName, MemberName, UniqueName},
     zvariant::{ObjectPath, Str, Type},
-    Error, MatchRuleBuilder, MessageType, Result,
+    Error, MessageType, Result,
 };
 
+mod builder;
+pub use builder::MatchRuleBuilder;
```

Comment:
The actual builder struct should no longer need the type prefix (e.g this should be just `match_rule::Builder`. For backwards compatibility we should provide an alias with existing name in the root.